### ISSUE TITLE
Intelmqctl fails to stop bots

### ISSUE THREAD

**Issue by putsi** [certtools/intelmq#411]
I noticed that the command below doesn't always stop the botnet properly and still deletes the PID-files. 

```
sudo -u intelmq /opt/intelmq/bin/intelmqctl --botnet stop --type json
```

Possible fix could be replacing the `SIGINT` with `SIGTERM` on the [Intelmqctl source](https://github.com/certtools/intelmq/blob/master/intelmq/bin/intelmqctl#L142) which will always stop the botnet properly.

**Comment by mauroasilva** [comment]
Thanks for the heads up... the ```SIGINT``` was meant to be a soft kill... but maybe we can think about something like a harsher kill for situations in which the bot takes too long to die. What do you think @aaronkaplan, @sebix, @robcza?

**Comment by sebix** [comment]
Sigint will be caught by the bot class as KeyboardInterrupt exception, and the bot class is terminated gracefully. We could do something similar for sigterm too.

intelmqctl can wait for a maximum time of e.g. 5 seconds after sending the signal an print an error message if the bot is still running. (But we use the pid file to determine this AFAIR) Then we can provide an optional flag for sigterm.

**Comment by aaronkaplan** [comment]
@putsi can you elaborate maybe what bot did not terminate?

**Comment by putsi** [comment]
If I remember correctly, I had deduplicator, filter, fileoutput and couple custom http-based collectors and parsers. None of those terminated properly.

**Comment by aaronkaplan** [comment]
> On 22 Dec 2015, at 13:32, Jarmo Puttonen <notifications@github.com> wrote:
> 
> If I remember correctly, I had deduplicator, filter, fileoutput and couple custom http-based collectors and parsers. None of those terminated properly.

thx

**Comment by aaronkaplan** [comment]
@putsi can you pls. re-test? there is a new intelmqctl

**Comment by sebix** [comment]
[This](https://github.com/certtools/intelmq/commit/4ad841ab1476e9a0f06472fa28d534ce3685de0f#diff-ee1e710d1c6b6b7301b04e00dbdefd42) only fixed a bug in the function reading the logs for python > 3.2. I don't think this will affect the termination of bots.

**Comment by putsi** [comment]
Well the SIGINT works now on my installation, but it might be due to changes in bots. I am currently using 4 HTTP-based collectors, 3 custom parsers, 2 custom filter experts and 1 custom output.

**Comment by aaronkaplan** [comment]
can this be closed now?

**Comment by dmth** [comment]
What the state of this issue?

**Comment by sebix** [comment]
We can't send SIGTERM currently. Probably this can be solved with systemd unit's too? #693

Concerning the pidfiles I created #710 (had this in mind for some weeks already)

**Comment by dmth** [comment]
I've seen this problem a lot last week. IMHO this should have a higher priority.

**Comment by sebix** [comment]
Once we use external libs to manage processes, like systemd or supervisord, we don't have this problem anymore.

So IMO #693 obsoletes this issue.

**Comment by dmth** [comment]
Nevertheless, It's still a problem.

Both, Intelmqctl and the manager show bots as stopped, nevertheless the remain running. Users need to be made aware, that the Bots could not be stopped, and they have to stop the processes on their own.

**Comment by wagner-certat** [comment]
Fixed by #871
